### ISSUE TITLE
Crash in timeunits.py. ZeroDivisionError: division by zero

### ISSUE THREAD

**Issue by kfhdk** [time4tea/gopro-dashboard-overlay#78]
I got a crash and a quick'n'dirty fix
I can provide a short video and gpx file

```
$ gopro-dashboard.py --gpx 'Morning_Ride (2).gpx' GH010049-rotate-cut.MP4 GH010049-rotate-dashboard-delme.mp4
Starting gopro-dashboard version 0.61.0
Timer(loading timeseries - Called: 1, Total: 0.60145, Avg: 0.60145, Rate: 1.66)
Timer(program - Called: 1, Total: 0.60150, Avg: 0.60150, Rate: 1.66)
Traceback (most recent call last):
  File "/usr/local/bin/gopro-dashboard.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/kfh/debs/gopro-dashboard-overlay/bin/gopro-dashboard.py", line 132, in <module>
    frame_meta = framemeta_from(input_file, metameta=stream_info.meta, units=units)
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/framemeta.py", line 243, in framemeta_from
    return parse_gopro(gpmd_from, units, metameta)
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/framemeta.py", line 233, in parse_gopro
    gps_frame_meta = gps_framemeta(gopro_meta, units, metameta=metameta)
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/framemeta.py", line 191, in gps_framemeta
    meta.accept(
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/gpmd.py", line 311, in accept
    item.accept(visitor)
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/gpmd.py", line 189, in accept
    i.accept(container_visitor)
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/gpmd.py", line 191, in accept
    container_visitor.v_end()
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/gpmd_visitors_gps.py", line 87, in v_end
    self._on_end(GPS5Components(
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/gpmd_visitors_gps.py", line 111, in <lambda>
    on_end=lambda c: self._converter(self._counter, c)
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/gpmd_visitors_gps.py", line 17, in convert
    sample_time_calculator = self._frame_calculator.next_packet(
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/gpmd_calculate.py", line 22, in next_packet
    time_per_sample = timeunits(millis=1001) / num_samples
  File "/home/kfh/debs/gopro-dashboard-overlay/gopro_overlay/timeunits.py", line 33, in __truediv__
    return Timeunit(self.us / other)
ZeroDivisionError: division by zero
```

I don't know what is should do, but the following "works".
```
diff --git a/gopro_overlay/timeunits.py b/gopro_overlay/timeunits.py
index dc0ed2e..cdf2619 100644
--- a/gopro_overlay/timeunits.py
+++ b/gopro_overlay/timeunits.py
@@ -30,7 +30,10 @@ class Timeunit:
         if type(other) == type(self):
             return self.us / other.us
         elif type(other) == int:
-            return Timeunit(self.us / other)
+            if other > 0:
+                return Timeunit(self.us / other)
+            else:
+                return Timeunit(self.us)
         raise NotImplementedError(f"Unsupported / between {type(self)} and {type(other)}")
 
     def __eq__(self, other):
```

**Comment by time4tea** [comment]
Thanks for reporting! - If you do have an example file that causes the error - I'd really appreciate it if you could share it with me. I usually suggest using wetransfer - to gopro-overlay (at) time4tea.net - If its not possible then of course absolutely no problem.

I'd like to see why we get zero at this point. I don't really like to do "defensive coding" unless we can show that this is definitely expected to get zero here (although definitely appreciate it is a good workaround for your right now)- it might just be hiding a different bug somewhere else.

**Comment by time4tea** [comment]
Thanks for the file! - I'll take a look :+1:

**Comment by time4tea** [comment]
Seems that GPS5 packets can indeed have zero items in.

**Comment by time4tea** [comment]
Fixed in 0.66.0
